htdocs/route.py

```diff
@@ -215,13 +243,25 @@ def edit_basic_file():
 
     status_message = ''
 
-    edit_file_path = tipi_disk_base + '/' + request.args.get('file')
-    rp = request.args.get('rp')
+    if request.args.get('rp'):
+        rp = request.args.get('rp')
+    elif request.form.get('rp'):
+        rp = request.form.get('rp')
+
+    if request.form.get('file_name'):
+        edit_file_path = tipi_disk_base + '/' + request.form.get('file_name')
+    elif request.args.get('file_name'):
+        edit_file_path = tipi_disk_base + '/' + request.args.get('file_name')
+
+    
     
     if request.form.get('action') == 'save_basic_file' or request.args.get('action') == 'save_basic_file':
+        eprint("action is save_basic_file")
         file_contents = request.form.get('file_contents')
         
-        file_name = request.form.get('file_name')  # Yeah, need to sanitize this, esp if this stuff runs as root!
+        file_name = request.form.get('file_name')  # Need to sanitize this and other user-supplied fields.
+        
+        eprint("file_name: ", file_name)
  
         bas_tmp_file = '/tmp/' + str(uuid.uuid4()) + '.tmp'
```

Comment:
I assume you are using UUIDs to allow concurancy...  these /tmp files need to be cleaned up so /tmp doesn't fill up.

Reply:
Correct. Normally I handle stuff like this with a crontab with something like this:
0 1 * * *	find /tmp/*.tmp -mtime +10 | xargs rm -f >>/dev/null 2>&1

Since TIPI is an appliance and not a server, perhaps the above plus a line like also:
@reboot find /tmp/*.tmp -mtime +10 | xargs rm -f >>/dev/null 2>&1

The alternative is to make tmp file deletion part of the code that creates them. I like the cron approach but maybe you have a preference?